src/java.base/share/classes/java/util/TreeMap.java

```diff
@@ -3322,16 +3311,12 @@ public int characteristics() {
         @Override
         public Comparator<Map.Entry<K, V>> getComparator() {
             // Adapt or create a key-based comparator
-            if (tree.comparator != null) {
-                return Map.Entry.comparingByKey(tree.comparator);
-            }
-            else {
-                return (Comparator<Map.Entry<K, V>> & Serializable) (e1, e2) -> {
```

Comment:
I think this cast hints compiler to bootstrap lambda meta factory with extra Serializable marker interface. Meanwhile, Comparator.naturalOrder does not implement Serializable.

Reply:
From one point this is obviously a change in behaviour, from another the spec of `Spliterator.getComparator()` says nothing about serializability of returned comparator and the actual value is hardly ever going to be serialized.

Reply:
The only new comparator that is returned here is the one returned by `Map.Entry.comparingByKey()`, which is also `Serializable`.

The only place `Comparator.naturalOrder()` is newly used in this PR are the `final` `package‑private` `getEntry` and `getEntryUsingComparator` methods.

Reply:
Also, `Comparator.naturalOrder()` is implemented using an `enum`, which is naturally `Serializable`:
https://github.com/openjdk/jdk/blob/0c13d66622a8c2be654bb867aa8c17421d1557ca/src/java.base/share/classes/java/util/Comparators.java#L42-L59